lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3294,18 +3294,48 @@ private function performCallbackOnCachedPersister(callable $callback)
         }
     }
 
+    private function dispatchFlushEntityListener($event)
+    {
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
+            foreach ($this->identityMap as $className => $entities) {
+                $hash = (isset($this->entityIdentifiers[$oid], $this->entityIdentifiers[$oid]['id'])) ? $this->entityIdentifiers[$oid]['id'] : null;
+                if ($hash) {
+                    $hash = (string) $hash;
+                    if (isset($this->identityMap[$className][$hash])) {
+                        $entity = $this->identityMap[$className][$hash];
+                        if (spl_object_hash($entity) === $oid) {
+                            $entityClass = get_class($entity);
+                            $class = $this->em->getClassMetadata($entityClass);
+                            if (!isset($subscribedSystems[$entityClass])) {
+                                $subscribedSystems[$entityClass] = $this->listenersInvoker->getSubscribedSystems($class, $event) & ~ListenersInvoker::INVOKE_MANAGER;
+                            }
+                            $invoke = $subscribedSystems[$entityClass];
+                            if (ListenersInvoker::INVOKE_NONE !== $invoke) {
```

Comment:
Wouldn't a return close the function and not allow it to continue to the next Entity Listener?